src/main/java/com/instaclick/pentaho/plugin/amqp/AMQPPlugin.java

```diff
@@ -100,6 +162,14 @@ public boolean processRow(StepMetaInterface smi, StepDataInterface sdi) throws K
 
                 setOutputDone();
 
+                //now output rowset closed, so we can wait for our confirmation steps
+                if ( data.activeConfirmation )  {
+                    while (!isStopped() && data.watchedConfirmStep.size() > 0) {    
+                        Thread.sleep(500);

```

Comment:
But it cant take longer than 500 to finish and remove from ``data.watchedConfirmStep`` right ?
 
Cant we just re-use the ``transFinished`` to ack/nack the messages at the end of the transformation ?